packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -766,7 +775,10 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
       if (widget.searchCallback != null) {
         currentHighlight = widget.searchCallback!(filteredEntries, _localTextEditingController!.text);
       } else {
-        currentHighlight = search(filteredEntries, _localTextEditingController!);
+        final bool shouldUpdateCurrentHighlight = _shouldUpdateCurrentHighlight(filteredEntries, _localTextEditingController!);
+        if(shouldUpdateCurrentHighlight){
```

Comment:
To fix the linux analyze test:
```suggestion
        if (shouldUpdateCurrentHighlight) {
```